design-proposals/flavors-and-preferences.md

```diff
@@ -0,0 +1,408 @@
+    * Features `VirtualMachineFlavorSpec.CPU.Features`
+    * isolateEmulatorThread `VirtualMachineFlavorSpec.CPU.isolateEmulatorThread`
+    * Model `VirtualMachineFlavorSpec.CPU.Model`
+    * [NUMA](http://kubevirt.io/api-reference/main/definitions.html#_v1_watchdog) `VirtualMachineFlavorSpec.CPU.NUMA`
+    * [Realtime](http://kubevirt.io/api-reference/main/definitions.html#_v1_realtime) `VirtualMachineFlavorSpec.CPU.Realtime`
+  * [Devices](http://kubevirt.io/api-reference/main/definitions.html#_v1_devices)
+    * autoattachGraphicsDevice `VirtualMachinePreferencesSpec.Devices.autoattachGraphicsDevice`
+    * autoattachMemBalloon `VirtualMachinePreferencesSpec.Devices.autoattachMemBalloon`
+    * autoattachPodInterface `VirtualMachinePreferencesSpec.Devices.autoattachPodInterface`
+    * autoattachSerialConsole `VirtualMachinePreferencesSpec.Devices.autoattachSerialConsole`
+    * blockMultiQueue `VirtualMachineFlavorSpec.Devices.blockMultiQueue`
+    * clientPassthrough None
+    * disableHotplug `VirtualMachinePreferencesSpec.Devices.disableHotplug`
+    * [Disks](http://kubevirt.io/api-reference/main/definitions.html#_v1_disk)
+      * [blockSize](http://kubevirt.io/api-reference/main/definitions.html#_v1_blocksize) `VirtualMachinePreferencesSpec.Devices.PreferredBlockSize`
+      * bootOrder None
+      * cache `VirtualMachinePreferencesSpec.Devices.PreferredBlockSize`
+      * [Cdrom](http://kubevirt.io/api-reference/main/definitions.html#_v1_cdromtarget)
+        * Bus `VirtualMachinePreferencesSpec.Devices.PreferredCdRomBus`
+        * Readonly None
+        * Tray None
+      * dedicatedIOThread `VirtualMachineFlavorSpec.Devices.dedicatedIOThread`
+      * [Disk](http://kubevirt.io/api-reference/main/definitions.html#_v1_disktarget)
+        * Bus `VirtualMachinePreferencesSpec.Devices.PreferredDiskBus`
+        * pciAddress None
+        * Readonly None
+      * io `VirtualMachineFlavorSpec.Devices.PreferredIOMode`
+      * [Lun](http://kubevirt.io/api-reference/main/definitions.html#_v1_luntarget)
+        * Bus `VirtualMachinePreferencesSpec.Devices.PreferredLunBus`
+        * readOnly None
+      * name None
+      * serial None
+      * shareable None
+      * tag None
+      * [Filesystems](http://kubevirt.io/api-reference/main/definitions.html#_v1_filesystem)
+        * Name None
+        * virtiofs None
+      * [Gpus](http://kubevirt.io/api-reference/main/definitions.html#_v1_gpu)
+        * deviceName `VirtualMachineFlavorSpec.Devices.GPUs`
+        * name None
+        * tag `VirtualMachineFlavorSpec.Devices.GPUs`
+        * virtualGPUOptions `VirtualMachinePreferencesSpec.Devices.GPUs.virtualGPUOptions`
+      * [HostDevices](http://kubevirt.io/api-reference/main/definitions.html#_v1_hostdevice)
+        * deviceName `VirtualMachineFlavorSpec.Devices.HostDevices`
+        * name None
+        * tag  `VirtualMachineFlavorSpec.Devices.HostDevices`
```

Comment:
I am unsure.

Reply:
ACK, I thought providing a tag would help users identify devices easier but we can add this in later if requried.